src/AuthenticationService.php

```diff
@@ -227,7 +227,7 @@ public function clearIdentity(ServerRequestInterface $request, ResponseInterface
     {
         foreach ($this->_authenticators as $authenticator) {
             if ($authenticator instanceof PersistenceInterface) {
-                $authenticator->clearIdentity($request);
+                $authenticator->clearIdentity($request, $response);
```

Comment:
@lorenzo we need the request as well, because we clear the identity attribute from it. Or will keep this the reference of the passed response?

```php
    public function clearIdentity(ServerRequestInterface $request, ResponseInterface $response)
    {
        foreach ($this->_authenticators as $authenticator) {
            if ($authenticator instanceof PersistenceInterface) {
                $response = $authenticator->clearIdentity($request, $response);
            }
        }

        return $request->withoutAttribute('identity');
    }
```